src/control.f90

```diff
@@ -824,6 +826,23 @@ subroutine md_run (fixed_potential, vary_mu, total_energy)
 
     do iter = i_first, i_last ! Main MD loop
        mdl%step = iter
+       
+       ! At a given time step, if the difference between T_ext and the final temperature is larger than 
+       ! the temperature step (with 5% margin), then update T_ext 
+       ! Temperature evolves linearly from temp_ion to temp_ion_end over MDn_steps steps
+       ! Stop when target temperature has been reached
+       if  (abs(mdl%T_ext - temp_ion_end) > 0.95*abs((temp_ion_end-temp_ion)/dble(MDn_steps))) then
+         mdl%T_ext = temp_ion+(dble(iter-1)/dble(MDn_steps))*(temp_ion_end-temp_ion)
```

Comment:
Can we define `(temp_ion_end - temp_ion)/real(MDn_steps,double)` as a variable such as `temp_change_step`? Please use `real( ,double)` rather than `dble()` for portability.

Reply:
Sorry for cutting in. But, I also would like to make this option of changing temperature available soon.
For the updating of target temperature(`mdl%T_ext`),
I have tested some melt-quench calculations with the following way.  It will shift the target temperature from `mdl%T_ext` to `temp_ion_end` with `temp_change_step` K for each md step. 
```
       temp_diff = temp_ion_end - mdl%T_ext
       if (abs(temp_diff) > temp_change_step) then
         mdl%T_ext = mdl%T_ext + temp_change_step * abs(temp_diff)/temp_diff
       else
         mdl%T_ext = temp_ion_end
       end if
```

Reply:
Than you for your comment. 
I would like to point out or ask the following:

- `temp_change_step` may be negative (ex: cooling) so the condition `(abs(temp_diff) > temp_change_step)`. Thus we should also use the absolute value of the temperature step in the condition. 
- If we restart the calculation, will the temperature be appropriately set by your code snippet? It had some issues with it before, so this is why I set the thermostat temperature directly as a linear function of the current step number `iter`.

Reply:
> * `temp_change_step` may be negative (ex: cooling) so the condition `(abs(temp_diff) > temp_change_step)` would be always true. Thus we should also use the absolute value of the temperature step in the condition.

I think a positive value of `temp_change_step` is good, because `temp_ion_end` gives a direction from current temperature for cooling or heating.

> * If we restart the calculation, will the temperature be appropriately set by your code snippet? It had some issues with it before, so this is why I set the thermostat temperature directly as a linear function of the current step number `iter`.

In my test, when restart, I set `AtomMove.ReadVelocity ` to True, and update `temp_ion`, so CQ will read velocity and make thermostat from this `temp_ion`. 
In develop branch, `iter` number will restart from beginning. 

> * There is a difference in behaviour  between the current implementation and the proposed one for simulations with restarts. What should happen when the final temperature is reached?
>   
>   1. Exit as soon as the temperature reaches `temp_ion_end`? The simulation then run exactly the number of steps required by the input file, regardless of whether a restart has occurred.
>   2. Continue and maintain the temperature at `temp_ion_end`? The number of steps actually run will depend on when restarts have occurred.

For a cooling or heating process, after `mdl%T_ext`  reaches `temp_ion_end`,  it may be good to exit.
which means after completing the rest part in this loop.